pkg/dataplane/server/listener.go

```diff
@@ -0,0 +1,89 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/clusterlink-net/clusterlink/pkg/controlplane/api"
+	dpapi "github.com/clusterlink-net/clusterlink/pkg/dataplane/api"
+)
+
+// DeleteListener deletes the listener to an imported service
+func (d *Dataplane) DeleteListener(name string) {
+	d.listenerChan[name] <- true
+}
+
+// CreateListener starts a listener to an imported service
+func (d *Dataplane) CreateListener(name, ip string, port uint32) {
+	listenTarget := ip + ":" + strconv.Itoa(int(port))
+	d.listenerChan[name] = make(chan bool)
+	d.logger.Infof("Starting a listener for imported service %s at %s.", name, listenTarget)
+	acceptor, err := net.Listen("tcp", listenTarget)
+	if err != nil {
+		d.logger.Infof("Error Listen to port %v", err)
+		return
+	}
+	go func() {
+		if err = d.serveEgressConnections(name, acceptor); err != nil {
+			d.logger.Errorf("Failed to serve egress connection on %s: %+v.", listenTarget, err)
+		}
+	}()
+	<-d.listenerChan[name]
```

Comment:
what is the channel synchronizing?

Reply:
It runs and waits until set by deleteListener